### PR TITLE
feat(ld-card-stack): allow cards to be displayed as a stack of cards

### LINKED ISSUE #409: Allow cards to be displayed as a stack of cards
With many cards and little screen estate it would be nice to stack cards in a way that the user only sees one card fully at a time while having a visual hint that there are more cards available.

As a follow up, it would be nice to be able to add, remove, shift cards in the stack with visual transitions.

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

With many cards and little screen estate it would be nice to stack cards in a way that the user only sees one card fully at a time while having a visual hint that there are more cards available. This PR introduces a new component which allows to stack multiple cards within its main slot. For now it is a quite stupid / simple component which does not include any interaction features.

Resolves #409

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] unit tests
- [x] e2e tests
- [x] accessibility tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css [hunk 1]
```diff
@@ -0,0 +1,78 @@
+:host,
+.ld-card-stack {
+  --ld-card-stack-translate-amount-x: var(--ld-sp-4);
+  --ld-card-stack-translate-amount-y: var(--ld-sp-4);
```
Comment (renet) [inline src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css:4]: I think this is a little too tight. I'd rather recommend twice the space, as it's hard for the eye to see the stacking details / it looks a little fuzzy due to the high details/lines with low contrasts.

File: src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  --ld-card-stack-translate-direction-x: -1;
```
Comment (renet) [inline src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css:5]: Why not also add the option to change the y direction, as well?

File: src/liquid/components/ld-card/ld-card.tsx [hunk 3]
```diff
@@ -23,6 +26,12 @@ export class LdCard {
   /** The size prop effects the card padding only. */
   @Prop() size?: 'sm'
 
+  componentWillLoad() {
+    if (this.el.parentElement.tagName === 'LD-CARD-STACK') {
+      this.el.setAttribute('role', 'listitem')
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-card/ld-card.tsx:33]: This is the simplest way to achieve what is needed here. If I wanted to move this logic to the `ld-card-stack` component, I would need to implement a mutation observer there, which would be overkill. I think it doesn't hurt much here.

File: src/liquid/components/ld-card/ld-card-stack/readme.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
+---
+eleventyNavigation:
+  key: Card Stack
+  parent: Card
+layout: layout.njk
+title: Card Stack
+permalink: components/ld-card/ld-card-stack/
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-card/ld-card-stack/readme.md:7]: Yes, I know. However, the `ld-card-stack` kind of extends the feature set of the `ld-card` component. It doesn't work or has any sense without the `ld-card` component, so listing it next to all other components wouldn't make much sense.

File: src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css (hunk 1 above)
Comment (borisdiakur) [reply src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css:4]: I've used [this](https://docs.merck.design/?selectedKind=Elements%2FCard&selectedStory=stacked&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs&knob-option%204=Option%204&knob-label%202%20value=Value%20%2F%20Number%202&knob-title=Title%20Name&knob-content=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel&knob-author=Neville%20Brody&knob-primary%20headline=We%27ve%20been%20around%20for%20350%20years%2C%20yet%20our%20majority%20owners%20are%20still%20the%20descendants%20of%20Friedrich%20Jacob%20Merck.&knob-title%200=Section%20title&knob-title%201=Section%20title&knob-content%200=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel%20fringilla%20est%20ullamcorper%20eget%20nulla%20facilisi%20etiam%20dignissim%20diam%20quis%20enim%20lobortis%20scelerisque%20fermentum%20dui%20faucibus%20in%20ornare%20quam%20viverra%20orci%20sagittis%20eu%20volutpat%20odio%20facilisis%20mauris%20sit%20amet%20massa%20vitae%20tortor%20condimentum%20lacinia%20quis%20vel%20eros%20donec%20ac%20odio%20tempor%20orci&knob-title%202=Section%20title&knob-content%201=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel%20fringilla%20est%20ullamcorper%20eget%20nulla%20facilisi%20etiam%20dignissim%20diam%20quis%20enim%20lobortis%20scelerisque%20fermentum%20dui%20faucibus%20in%20ornare%20quam%20viverra%20orci%20sagittis%20eu%20volutpat%20odio%20facilisis%20mauris%20sit%20amet%20massa%20vitae%20tortor%20condimentum%20lacinia%20quis%20vel%20eros%20donec%20ac%20odio%20tempor%20orci&knob-title%203=Section%20title&knob-content%202=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel%20fringilla%20est%20ullamcorper%20eget%20nulla%20facilisi%20etiam%20dignissim%20diam%20quis%20enim%20lobortis%20scelerisque%20fermentum%20dui%20faucibus%20in%20ornare%20quam%20viverra%20orci%20sagittis%20eu%20volutpat%20odio%20facilisis%20mauris%20sit%20amet%20massa%20vitae%20tortor%20condimentum%20lacinia%20quis%20vel%20eros%20donec%20ac%20odio%20tempor%20orci&knob-title%204=Section%20title&knob-quotation=Digital%20design%20is%20like%20painting%2C%20except%20the%20paint%20never%20dries%20ever%20again.&knob-content%203=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel%20fringilla%20est%20ullamcorper%20eget%20nulla%20facilisi%20etiam%20dignissim%20diam%20quis%20enim%20lobortis%20scelerisque%20fermentum%20dui%20faucibus%20in%20ornare%20quam%20viverra%20orci%20sagittis%20eu%20volutpat%20odio%20facilisis%20mauris%20sit%20amet%20massa%20vitae%20tortor%20condimentum%20lacinia%20quis%20vel%20eros%20donec%20ac%20odio%20tempor%20orci&knob-title%205=Section%20title&knob-content%204=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel%20fringilla%20est%20ullamcorper%20eget%20nulla%20facilisi%20etiam%20dignissim%20diam%20quis%20enim%20lobortis%20scelerisque%20fermentum%20dui%20faucibus%20in%20ornare%20quam%20viverra%20orci%20sagittis%20eu%20volutpat%20odio%20facilisis%20mauris%20sit%20amet%20massa%20vitae%20tortor%20condimentum%20lacinia%20quis%20vel%20eros%20donec%20ac%20odio%20tempor%20orci&knob-content%205=lorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit%20ut%20aliquam%2C%20purus%20sit%20amet%20luctus%20venenatis%2C%20lectus%20magna%20fringilla%20urna%2C%20porttitor%20rhoncus%20dolor%20purus%20non%20enim%20praesent%20elementum%20facilisis%20leo%2C%20vel%20fringilla%20est%20ullamcorper%20eget%20nulla%20facilisi%20etiam%20dignissim%20diam%20quis%20enim%20lobortis%20scelerisque%20fermentum%20dui%20faucibus%20in%20ornare%20quam%20viverra%20orci%20sagittis%20eu%20volutpat%20odio%20facilisis%20mauris%20sit%20amet%20massa%20vitae%20tortor%20condimentum%20lacinia%20quis%20vel%20eros%20donec%20ac%20odio%20tempor%20orci&knob-label%202%20name=Label%202&knob-option%201=Option%201&knob-label%201%20name=Label%201&knob-option%202=Option%202&knob-label%201%20value=Value%20%2F%20Number%201&knob-option%203=Option%203) as orientation.

File: src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css (hunk 2 above)
Comment (borisdiakur) [reply src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css:5]: I've talked to Nic and he said he would only need the three options.

File: src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css (hunk 1 above)
Comment (renet) [reply src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css:4]: Doesn't mean that this is a good solution. :D Maybe we could ask Nic about his opinion. If he agrees with you, I'll be fine with it.

**Review by renet — APPROVED** [review]
Feel free to merge, if you disagree with asking Nic or already talked to him about the spacing.

File: src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css (hunk 1 above)
Comment (borisdiakur) [reply src/liquid/components/ld-card/ld-card-stack/ld-card-stack.css]: I have talked to Nic. He also suggested to increase the distance, which I just did.

**Pull request merged**